chef/cookbooks/network/recipes/default.rb

```diff
@@ -209,6 +209,16 @@ def get_datapath_id_for_ovsbridge(bridge)
     ifs[bond.name]["addresses"] ||= Array.new
     ifs[bond.name]["slaves"] = Array.new
     base_ifs.each do |i|
+      # If the interface is configured, wicked will helpfully re-add the IP
+      # when the interface is downed and then brought back up -- which is
+      # required, since interfaces which are up can not be added as slaves.
+      # To stop this happening, remove the configuration file and reload the
+      # interface.
+      if ::File.exist?("/etc/sysconfig/network/ifcfg-#{i.name}")
```

Comment:
Hmmm, maybe we check if the configuration file contains an IP address, and then perform this block? That would solve the issue we're seeing, and mean it's a no-op for subsequent chef-client runs.